playbooks/server-install.yml

```diff
@@ -213,28 +214,27 @@
           - ca-certificates
           - libnss-libvirt
         state: present
-        update_cache: yes
-    
-    - name: Install tailscale
-      shell: curl -fsSL https://tailscale.com/install.sh | sh
-      when: tailscale_auth_key is defined
-        
-    - name: Start tailscale
-      command: tailscale up --authkey {{ tailscale_auth_key }} --ssh
-      when: tailscale_auth_key is defined
-      no_log: true
+        allow_downgrade: yes
 
     - name: Enable and start libvirtd service
       service:
         name: libvirtd
         enabled: yes
         state: started
 
-    - name: Enable and start the default libvirt network
-      shell: |
-        virsh net-start default
-        virsh net-autostart default
+    - name: Check libvirt 'default' network state
+      command: virsh net-info default
+      register: net_info
       changed_when: false
+      failed_when: false
+
+    - name: Start the default libvirt network if it is not active
+      command: virsh net-start default
+      when: "(net_info.rc == 0) and ('Active: no' in net_info.stdout)"
+
+    - name: Enable autostart for the default libvirt network
+      command: virsh net-autostart default
+      when: "(net_info.rc == 0) and ('Autostart: no' in net_info.stdout)"
```

Comment:
shouldn't this run always? If it's already active wouldn't it not set it to autostart?